files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/appList.js

```diff
@@ -333,6 +335,8 @@ class AppList {
             // Abort the remove if the window is just changing workspaces, window
             // should always remain indexed on all workspaces while its mapped.
             // if (!metaWindow.showing_on_its_workspace()) return;
+            if ((!metaWindow.showing_on_its_workspace() && metaWindow.minimize())
+            || metaWindow.has_focus() && metaWorkspace.index() !== global.screen.get_active_workspace_index()) return;
             this.state.removingWindowFromWorkspaces = true;
             this.state.trigger('removeWindowFromAllWorkspaces', metaWindow);
             return;
```

Comment:
Not sure, just a suggestion: this one probably looks prettier, because it only uses one return in the block and (as to me) it's easier to understand this piece as "if we should abort then (if condition then do something); stop" than "if we should abort then (if condition then stop); do something; stop"
```suggestion
            if ((metaWindow.showing_on_its_workspace() || !metaWindow.minimize())
            && (!metaWindow.has_focus() || metaWorkspace.index() === global.screen.get_active_workspace_index()))
            {
                this.state.removingWindowFromWorkspaces = true;
                this.state.trigger('removeWindowFromAllWorkspaces', metaWindow);
            }
            return;
```

Reply:
@kolayne thanks. I'm was just trying to solve it with minimal changes :) but I see your point and seems like a nice suggestion. Works fine here.